tests/alerts_tests.py

```diff
@@ -41,114 +49,267 @@
+                sql="$%^&",
+                alert_id=db.session.query(Alert).filter_by(label="alert_3").one().id,
+                database_id=example_database_id,
             ),
-            Alert(
-                id=2,
-                label="alert_2",
-                active=True,
-                crontab="*/1 * * * *",
+            SQLObserver(
                 sql="SELECT 55",
-                alert_type="email",
-                slice_id=slice_id,
-                recipients="recipient1@superset.com",
-                slack_channel="#test_channel",
-                database_id=database_id,
+                alert_id=db.session.query(Alert).filter_by(label="alert_4").one().id,
+                database_id=example_database_id,
             ),
-            Alert(
-                id=3,
-                label="alert_3",
-                active=False,
-                crontab="*/1 * * * *",
-                sql="UPDATE 55",
-                alert_type="email",
-                slice_id=slice_id,
-                database_id=database_id,
+            SQLObserver(
+                sql="SELECT 'test_string' as string_value",
+                alert_id=db.session.query(Alert).filter_by(label="alert_5").one().id,
+                database_id=example_database_id,
+            ),
+            SQLObserver(
+                sql="SELECT null as null_result",
+                alert_id=db.session.query(Alert).filter_by(label="alert_6").one().id,
+                database_id=example_database_id,
+            ),
+            SQLObserver(
+                sql="SELECT 30.0 as wage",
+                alert_id=db.session.query(Alert).filter_by(label="alert_7").one().id,
+                database_id=example_database_id,
+            ),
+            SQLObserver(
+                sql="SELECT first FROM test_table",
+                alert_id=db.session.query(Alert).filter_by(label="alert_8").one().id,
+                database_id=example_database_id,
+            ),
+            SQLObserver(
+                sql="SELECT first, second FROM test_table WHERE first = 1",
+                alert_id=db.session.query(Alert).filter_by(label="alert_9").one().id,
+                database_id=example_database_id,
             ),
-            Alert(id=4, active=False, label="alert_4", database_id=-1),
-            Alert(id=5, active=False, label="alert_5", database_id=database_id),
         ]
 
-        db.session.bulk_save_objects(alerts)
-        db.session.commit()
+        db.session.bulk_save_objects(observers)
         yield db.session
 
+        db.session.query(SQLObservation).delete()
+        db.session.query(SQLObserver).delete()
+        db.session.query(Validator).delete()
         db.session.query(AlertLog).delete()
         db.session.query(Alert).delete()
 
 
+def test_alert_observer(setup_database):
+    dbsession = setup_database
+
+    # Test SQLObserver with int SQL return
+    alert4 = dbsession.query(Alert).filter_by(label="alert_4").one()
```

Comment:
I would prefer a bit different setup as mentioned before:
```
NOTNULLCONFIG="""
  "type": "notnull"
"""
alert = create_alert(sql='....', config=NOTNULLCONFIG)
...
```

and create alert would create alert, observer and validator 

It will help to keep sql text & config closer to the test code & make it more readable